x-pack/legacy/plugins/siem/public/components/timeline/search_super_select/index.tsx

```diff
@@ -0,0 +1,257 @@
+import { getEmptyTagValue } from '../../empty_value';
+import { isUntitled } from '../../../components/open_timeline/helpers';
+import * as i18nTimeline from '../../../components/open_timeline/translations';
+import { SortFieldTimeline, Direction } from '../../../graphql/types';
+
+import * as i18n from './translations';
+
+const SearchTimelineSuperSelectGlobalStyle = createGlobalStyle`
+  .euiPopover__panel.euiPopover__panel-isOpen.timeline-search-super-select-popover__popoverPanel {
+    visibility: hidden;
+    z-index: 0;
+  }
+`;
+
+interface SearchTimelineSuperSelectProps {
+  isDisabled: boolean;
+  timelineId: string | null;
+  timelineTitle: string | null;
+  onTimelineChange: (timelineTitle: string, timelineId: string) => void;
+}
+
+const basicSuperSelectOptions = [
+  {
+    value: '-1',
+    inputDisplay: i18n.DEFAULT_TIMELINE_TITLE,
+  },
+];
+
+const basicSelectableOptions = [
+  {
+    description: i18n.DEFAULT_TIMELINE_DESCRIPTION,
+    label: i18n.DEFAULT_TIMELINE_TITLE,
+    id: null,
+    title: i18n.DEFAULT_TIMELINE_TITLE,
+  } as Option,
+];
+
+const OriginalPageSize = 3;
+const PopoverHeight = 90;
+const TimelineItemHeight = 50;
+const SearchTimelineSuperSelectComponent: React.FC<SearchTimelineSuperSelectProps> = ({
+  isDisabled,
+  timelineId,
+  timelineTitle,
+  onTimelineChange,
+}) => {
+  const [pageSize, setPageSize] = useState(OriginalPageSize);
+  const [heightTrigger, setHeightTrigger] = useState(0);
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [searchTimelineValue, setSearchTimelineValue] = useState('');
+  const [onlyFavorites, setOnlyFavorites] = useState(false);
+
+  const onSearchTimeline = useCallback(val => {
+    setSearchTimelineValue(val);
+  }, []);
+
+  const handleClosePopover = useCallback(() => {
+    setIsPopoverOpen(false);
+  }, []);
+
+  const handleOpenPopover = useCallback(() => {
+    setIsPopoverOpen(true);
+  }, []);
+
+  const handleOnToggleOnlyFavorites = useCallback(() => {
+    setOnlyFavorites(!onlyFavorites);
+  }, [onlyFavorites]);
+
+  const renderTimelineOption = useCallback((option, searchValue) => {
+    return (
+      <>
+        <EuiHighlight search={searchValue}>
+          {isUntitled(option) ? i18nTimeline.UNTITLED_TIMELINE : option.title}
+        </EuiHighlight>
+        <br />
+        <EuiTextColor color="subdued">
+          <small>
+            {option.description != null && option.description.trim().length > 0
+              ? option.description
+              : getEmptyTagValue()}
+          </small>
+        </EuiTextColor>
+      </>
+    );
+  }, []);
+
+  const handleTimelineChange = useCallback(options => {
+    const selectedTimeline = options.filter(
+      (option: { checked: string }) => option.checked === 'on'
+    );
+    if (selectedTimeline != null && selectedTimeline.length > 0 && onTimelineChange != null) {
+      onTimelineChange(selectedTimeline[0].title, selectedTimeline[0].id);
```

Comment:
`selectedTimeline[0].id` can be null -- `selectedTimeline` is an `any` so linter isn't showing it.

Reply:
it is expected to be null